cmd/fleet/handleStatus.go

```diff
@@ -8,31 +8,135 @@ import (
 	"context"
 	"encoding/json"
 	"net/http"
+	"time"
 
+	"github.com/elastic/fleet-server/v7/internal/pkg/apikey"
+	"github.com/elastic/fleet-server/v7/internal/pkg/bulk"
+	"github.com/elastic/fleet-server/v7/internal/pkg/cache"
+	"github.com/elastic/fleet-server/v7/internal/pkg/config"
+	"github.com/elastic/fleet-server/v7/internal/pkg/limit"
 	"github.com/elastic/fleet-server/v7/internal/pkg/logger"
+	"github.com/julienschmidt/httprouter"
 
 	"github.com/elastic/elastic-agent-client/v7/pkg/proto"
-	"github.com/julienschmidt/httprouter"
+	"github.com/rs/zerolog"
 	"github.com/rs/zerolog/log"
 )
 
-func (rt Router) handleStatus(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+const (
+	kStatusMod = "status"
+)
 
-	dfunc := cntStatus.IncStart()
-	defer dfunc()
+type AuthFunc func(*http.Request) (*apikey.ApiKey, error)
+
+type StatusT struct {
+	cfg    *config.Server
+	limit  *limit.Limiter
+	bulk   bulk.Bulk
+	cache  cache.Cache
+	authfn AuthFunc
+}
+
+type OptFunc func(*StatusT)
+
+func NewStatusT(cfg *config.Server, bulker bulk.Bulk, cache cache.Cache, opts ...OptFunc) *StatusT {
+	log.Info().
+		Interface("limits", cfg.Limits.StatusLimit).
+		Msg("Setting config status_limits")
+
+	st := &StatusT{
+		cfg:   cfg,
+		bulk:  bulker,
+		cache: cache,
+		limit: limit.NewLimiter(&cfg.Limits.StatusLimit),
+	}
+	st.authfn = st.authenticate
+
+	for _, opt := range opts {
+		opt(st)
+	}
+	return st
+}
+
+func WithAuthFunc(authfn AuthFunc) OptFunc {
+	return func(st *StatusT) {
+		if authfn != nil {
+			st.authfn = authfn
+		}
+	}
+}
+
+func (st StatusT) authenticate(r *http.Request) (*apikey.ApiKey, error) {
+	return authApiKey(r, st.bulk, st.cache)
```

Comment:
@aleksmaus I've looked at the implementation of `authApiKey`, this only verifies that the key is active and valid, we do not do any additional checks. I don't think the current information that the Key is protecting is crucial so in that case "it's okay". 

But it would be good to add some godoc here, so if anyone wants to use that authentication methods it knows that it has its limit.

Reply:
it's already there
https://github.com/elastic/fleet-server/blob/130056a170a3ca0b3b8ce4df12741ac40ae70c91/cmd/fleet/auth.go#L28